tripy/examples/resnet50/resnet50.ipynb

```diff
@@ -0,0 +1,483 @@
+    "            (3, 256, 64, (1, 1)),\n",
+    "            (4, 512, 128, (2, 2)),\n",
+    "            (6, 1024, 256, (2, 2)),\n",
+    "            (3, 2048, 512, (2, 2)),\n",
+    "        ]\n",
+    "        self.encoder = ResNetEncoder(layers_config)\n",
+    "\n",
+    "    def __call__(self, x: tp.Tensor) -> tp.Tensor:\n",
+    "        x = self.embedder(x)\n",
+    "        x = self.encoder(x)\n",
+    "        x = tp.avgpool(\n",
+    "            x, kernel_dims=(7, 7), stride=(7, 7)\n",
+    "        )  # output size will be (1, 1)\n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "class ResNetClassifier(tp.Module):\n",
+    "    def __init__(self, num_classes: int = 1000):\n",
+    "        \"\"\"Combines the backbone ResNetModel and a classifier head for\n",
+    "        outputting class probabilities.\"\"\"\n",
+    "        super(ResNetClassifier, self).__init__()\n",
+    "        self.resnet = ResNetModel()\n",
+    "        self.classifier = tp.Sequential(\n",
+    "            lambda x: tp.flatten(x, start_dim=1),\n",
+    "            tp.Linear(2048, num_classes),\n",
+    "        )\n",
+    "\n",
+    "    def __call__(self, x: tp.Tensor) -> tp.Tensor:\n",
+    "        features = self.resnet(x)\n",
+    "        return self.classifier(features)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Instantiate the Tripy model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tripy_model = ResNetClassifier()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load Weights to Tripy Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import ResNetForImageClassification\n",
+    "\n",
+    "# Load the pretrained ResNet50 model from Hugging Face\n",
+    "resnet_pretrained = ResNetForImageClassification.from_pretrained(\n",
+    "    \"microsoft/resnet-50\"\n",
+    ").to(\"cuda\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_pretrained_weights(model, pretrained_state_dict):\n",
+    "    model_state_dict_keys = set(model.state_dict().keys())\n",
+    "    model_dict = {}\n",
+    "\n",
+    "    # Convert each weight to a tp.Parameter and update the model's state dict\n",
+    "    for k, v in pretrained_state_dict.items():\n",
+    "        if \"num_batches_tracked\" in k:\n",
+    "            continue  # Skip num_batches_tracked since it is not needed for\n",
+    "            # inference (tp.BatchNorm does not support)\n",
+    "\n",
+    "        assert k in model_state_dict_keys\n",
+    "        model_dict[k] = tp.Parameter(v.contiguous())\n",
+    "\n",
+    "    model.load_state_dict(model_dict)\n",
+    "    return model\n",
```

Comment:
Since we're updating the model in-place, we don't need to return it here

Reply:
done